lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -1790,17 +1793,123 @@ class _DomTouchListWrapper<T> extends Iterable<T> {
 Iterable<T> createDomTouchListWrapper<T>(_DomTouchList list) =>
     _DomTouchListWrapper<T>._(list).cast<T>();
 
+@JS()
+@staticInterop
+class DomSymbol {}
+
+extension DomSymbolExtension on DomSymbol {
+  external Object get iterator;
+}
+
 @JS()
 @staticInterop
 class DomIntl {}
 
 extension DomIntlExtension on DomIntl {
+  // ignore: non_constant_identifier_names
+  external Object? get Segmenter;
+
   /// This is a V8-only API for segmenting text.
   ///
   /// See: https://code.google.com/archive/p/v8-i18n/wikis/BreakIterator.wiki
   external Object? get v8BreakIterator;
 }
 
+/// Similar to [js_util.callMethod] but allows for providing a non-string method
+/// name.
+T jsCallMethod<T>(Object o, Object method, [List<Object?> args = const <Object?>[]]) {
```

Comment:
Can this be moved to `safe_browser_api.dart` and wrapped as to not be publicly available to all of the engine code?

Reply:
If it's too difficult to move, can it stay private to this file?

Reply:
I understand the benefit of making it private, so I'll make it private.

Would you mind clarifying what's the benefit of moving it to `safe_browser_api.dart`?

Reply:
This function can execute arbitrary code. It's good to avoid ad hoc calls out to JS from all over the codebase. So libraries like `dom.dart` and `safe_browser_api.dart` should provide a safe subset of the API to the rest of the engine.

Reply:
I agree with making it private so it doesn't get called everywhere.

Does it make a difference if we put it in `dom.dart` vs `safe_browser_api.dart`? Like is there a special handling for that file or something?